ManifoldEM/quaternion.py

```diff
@@ -237,3 +238,24 @@ def calc_avg_pd(q, nS):
                          q[0, :]**2 + q[3, :]**2 - np.ones((1, nS)) / 2.0))
 
     return PDs
+
+
+def convert_euler_to_S2(euler_angles):
+    rotation = Rotation.from_euler('ZXZ', euler_angles, degrees=False).as_matrix()
+    rxz, ryz, rzz = rotation[:,:,-1].T
+    s2 = np.stack([-ryz,rxz,rzz], axis=1).T
+    return s2
+
+
+def collapse_to_half_space_euler_angles(euler_angles):
+    s2 = convert_euler_to_S2(euler_angles)
+    is_mirrored = s2[0,:] < 0.0
+    s2[:,is_mirrored] = -s2[:,is_mirrored]
+    return s2, is_mirrored
+
+
+def qs_to_spider(raw_qs):
```

Comment:
This function only works half the time...

Reply:
resolved with "alternate function". I'm still not sure what makes the spider convention the spider one... but the spider one is present in either the canonical or alternate return of `qs_to_spider_euler_angles`.

Reply:
since `q2Spider` does an unbounded fit, i think the convention chosen is effectively random. I've removed most of the alternate stuff from the code, and will get rid of all of it once we've proven that we can get the same results upstream. I don't think it's worth trying to predict since it shouldn't matter anyway...